assets/js/util/i18n.js

```diff
@@ -19,22 +19,189 @@
+			numberFormat( prepareForReadableLargeNumber( number ), number % 10 === 0 ? {} : withSingleDecimal )
+		);
+	}
+
+	// Numbers between 10,000 and 1,000,000 round normally and have no decimals
+	if ( 10000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in thousands.
+			__( '%sK', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ) )
+		);
+	}
+
+	// Numbers between 1,000 and 10,000 round normally and display a single decimal unless the decimal is 0.
+	if ( 1000 <= number ) {
+		return sprintf(
+			// translators: %s: an abbreviated number in thousands.
+			__( '%sK', 'google-site-kit' ),
+			numberFormat( prepareForReadableLargeNumber( number ), number % 10 === 0 ? {} : withSingleDecimal )
+		);
+	}
+
+	return numberFormat( number );
+};
+
+/**
+ * Formats a number with unit using the JS Internationalization Number Format API.
+ *
+ * In addition to the supported 'style' values of the lower-level `numberFormat` function, this function
+ * supports two additional 'style' values 'metric' and 'seconds'.
+ *
+ * Another differentiation in behavior is that by default the function will use 'metric' formatting instead
+ * of 'decimal' formatting.
+ *
+ * @since n.e.x.t
+ *
+ * @param {number|string}                     number    The number to format.
+ * @param {(Intl.NumberFormatOptions|string)} [options] Formatting options or unit shorthand.
+ *                                                      Possible shorthand values are '%', 's',
+ *                                                      or a currency code.
+ * @return {string} The formatted number with unit.
+ */
+export const numFmt = ( number, options = {} ) => {
+	// Cast parsable values to numeric types.
+	number = isFinite( number ) ? number : Number( number );
+
+	if ( ! isFinite( number ) ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'Invalid number', number, typeof number );
+		return null;
+	}
+
+	let formatOptions = {};
+
+	// Expand shorthand values for units.
+	if ( '%' === options ) {
+		formatOptions = {
+			style: 'percent',
+			maximumFractionDigits: 2,
+		};
+	} else if ( 's' === options ) {
+		formatOptions = {
+			style: 'seconds',
+		};
+	} else if ( !! options && typeof options === 'string' ) {
+		formatOptions = {
+			style: 'currency',
+			currency: options,
+		};
+	} else if ( typeof options === 'object' ) {
+		formatOptions = { ...options };
+	}
+
+	// Note: `metric` is our custom, default style.
+	const { style = 'metric' } = formatOptions;
+
+	if ( 'metric' === style ) {
+		return readableLargeNumber( number );
+	}
+
+	if ( 'seconds' === style ) {
+		return prepareSecondsForDisplay( number );
+	}
+
+	try {
+		return numberFormat( number, formatOptions );
+	} catch ( e ) {
+		if ( !! options && typeof options === 'string' ) {
+			// if the formatting doesn't work, return the concatenated options(assuming as a custom unit) and number
+			return `${ number }${ options }`;
+		}
+	}
```

Comment:
@felixarntz do we know what conditions where the fallback would actually be used here? I know it's the fallback but simply appending the unit isn't very i18n-friendly, however making a translatable string for generally appending an unknown unit to a number isn't i18n-friendly either.

At a minimum, we should at least return `numberFormat( number )` (with no options) to apply basic formatting to the number which shouldn't fail.

Also in this code block:
* `( e )` can be removed since it isn't used
* If `options` is falsy or not a string then the function would return `undefined`

Reply:
I don't think we need this. I believe the try catch was added here because in the original logic where this kinda came from, we were trying to format with `currency` style and if that failed using the fallback of concatenating. But now that we catch all other special shorthands before (`%` and `s`), we should be good without this. It will also show potential errors on our end explicitly.